aria/orchestrator/workflows/api/task.py

```diff
@@ -69,21 +73,38 @@ def __init__(self,
                  ignore_failure=None,
                  inputs=None,
                  plugin=None,
-                 runs_on=None):
+                 runs_on=None,
+                 dry=False):
         """
         Creates an operation task using the name, details, node instance and any additional kwargs.
-        :param name: the operation of the name.
+
+        :param name: the name of the operation.
         :param actor: the operation host on which this operation is registered.
         :param inputs: operation inputs.
         """
-        assert isinstance(actor, (model.Node,
-                                  model.Relationship))
+
+        assert isinstance(actor, (models.Node, models.Relationship))
         super(OperationTask, self).__init__()
+
+        if dry:
+            from ..dry import convert_to_dry
+            plugin, implementation, inputs = convert_to_dry(plugin, implementation, inputs)
+
+        # Coerce inputs
+        if inputs is None:
+            inputs = {}
+        else:
+            for k, v in inputs.iteritems():
+                if not isinstance(v, models.Parameter):
+                    inputs[k] = models.Parameter(name=k,
+                                                 type_name=full_type_name(v),
```

Comment:
the type here should in principle correspond to the one in template's operation's inputs, correct? i understand theres no validation on this at the moment but im asking whether theres any benefit to saving the type name of an arbitrary input